### PR TITLE
CI: Wait for pr db before running e2e tests

### LINKED ISSUE #747: CI: Check if backend is ready before running E2E tests
Currently there is no good technique implemented that waits for the backend before starting the tests.

### PR DESCRIPTION
**Pull request by 4ydan**
Fixes #747 

Not sure if this is a good solution. Open to suggestions.

## Basics

- [ ] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -9,6 +9,13 @@ def wait_for_db() {
     }
 }
 
+def wait_for_pr_db() {
+    retry(6) {
+        sleep(time: 5, unit: 'SECONDS')
+        sh 'curl https://pr.permaplant.net/api/config'
```
Comment (markus2330) [inline ci/Jenkinsfile:15]: To also fail unsuccessful responses from the server (5xx or similar):

```suggestion
        sh 'curl --fail https://pr.permaplant.net/api/config'
```

**Review by markus2330 — APPROVED** [review]
I think the solution is fine.

File: ci/Jenkinsfile (hunk 1 above)
Comment (4ydan) [reply ci/Jenkinsfile:15]: Perfect, thank you.

**Comment by 4ydan** [comment]
Ready to be merged.

**Pull request merged**